### PR TITLE
Cadc 9963 Updated caom2-tap and caom2-soda-server modules.

### PR DESCRIPTION
**Pull request by yeunga**
Updated CadcMastCutoutGenerator and SodaJobRunner.

### CODE REVIEW

File: caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java [hunk 1]
```diff
@@ -321,4 +306,21 @@ public URL toURL(int serialNum, URI uri, Cutout cutout, Map<String, List<String>
...
+        path.append("|text/plain");
+        path.append("|").append(errorMessage);
+        String msg = Base64.encodeString(path.toString());
+
+        // hack to get base url for soda service
+        URL serviceURL = reg.getServiceURL(sodaURI, Standards.SODA_SYNC_10, AuthMethod.ANON);
+        URL url = new URL(serviceURL.toExternalForm() + "/" + msg);
+        String surl = serviceURL.toExternalForm().replace("/sync", "/soda-echo");
+        surl = surl + "/" + msg;
+        URL loc = new URL(url.toExternalForm().replace("/sync", "/soda-echo"));
```
Comment (pdowler) [inline caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java]: this `new URL` ignores the `surl` put together above

File: caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java [hunk 2]
```diff
@@ -155,16 +155,14 @@ public URL toURL(int serialNum, URI uri, Cutout cutout, Map<String, List<String>
...
-                String msg = Base64.encodeString(path.toString());
+                return errorURL("NotFound: " + uri);
+            }
 
-                URL serviceURL = reg.getServiceURL(sodaURI, Standards.SODA_SYNC_10, AuthMethod.ANON);
-                URL url = new URL(serviceURL.toExternalForm() + "/" + msg);
-                URL loc = new URL(url.toExternalForm().replace("/sync", "/soda-echo"));
-                return loc;
+            StorageResolver resolver = artifactResolver.getStorageResolver(uri);
+            if (resolver instanceof CutoutGenerator) {
```
Comment (pdowler) [inline caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java:162]: the same errorURL response should be generated if the resolver is not a CutoutGenerator.

Also, you could keep the reference to the CutoutGenerator as a variable to use below and not have to get it again and check type around line~191. Then the UnsupportedOperation below (in the else) would be removed.

**Review by pdowler — CHANGES_REQUESTED** [review]

File: caom2-tap/src/main/java/ca/nrc/cadc/caom2ops/CadcMastCutoutGenerator.java [hunk 3]
```diff
@@ -81,44 +84,32 @@
  */
 public class CadcMastCutoutGenerator extends CadcMastResolver implements CutoutGenerator {
     private static final Logger log = Logger.getLogger(CadcMastCutoutGenerator.class);
+    private final CutoutGenerator cadcCutoutGenerator = new CadcCutoutGenerator();
```
Comment (pdowler) [inline caom2-tap/src/main/java/ca/nrc/cadc/caom2ops/CadcMastCutoutGenerator.java]: This needs to be a CadcCutoutGenerator("mast") ??

**Review by pdowler — CHANGES_REQUESTED** [review]

File: caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java (hunk 1 above)
Comment (yeunga) [reply caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java]: Fixed.

File: caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java (hunk 2 above)
Comment (yeunga) [reply caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java:162]: Updated.

File: caom2-tap/src/main/java/ca/nrc/cadc/caom2ops/CadcMastCutoutGenerator.java (hunk 3 above)
Comment (yeunga) [reply caom2-tap/src/main/java/ca/nrc/cadc/caom2ops/CadcMastCutoutGenerator.java]: Fixed.

**Review by pdowler — APPROVED** [review]

**Pull request merged**